src/comlink.ts

```diff
@@ -486,8 +487,7 @@ function createProxy<T>(
           }
           propProxyCache.clear();
           unregisterProxy(proxy);
-          releaseEndpoint(ep);
-          pendingListeners.clear();
```

Comment:
Is the issue that clearing the listeners here is premature because the `releaseEndpoint` call needs to access the listeners to resolve? If that's the case it looks like `releaseEndpoint` itself returns a promise so you could do a `.finally(() => pendingListeners.clear())` here instead of passing it to releaseEndpoint?

Or is it important to pass `pendingListeners` so it can be passed to `requestResponseMessage` ?

Reply:
> Or is it important to pass `pendingListeners` so it can be passed to `requestResponseMessage` ?

Yes, that's the important bit. If `pendingListeners` are not passed, the [resolver function is not found](https://github.com/foxglove/comlink/blob/ee0e6b8fc5b2fe3ece5287bc880df25ad33f7f8e/src/comlink.ts#L406-L415) and the `requestResponseMessage` promise will never settle.

Reply:
So why do we allow pendingListeners to be undefined? In what situation is that ok?

Reply:
This was necessary when [releasing the endpoint when the proxy is garbage collected](https://github.com/foxglove/comlink/blob/ee0e6b8fc5b2fe3ece5287bc880df25ad33f7f8e/src/comlink.ts#L454). However, thinking about this more, we should also have access to the `pendingListeners` map as it is tightly coupled to the endpoint now.

Reply:
I have added a commit that changes the `endpoint` and `pendingListeners` to be passed around in a single object (`EndpointWithPendingListeners`). There is probably a better name for that type/object...